<?xml version="1.0"?>

<launch>
  <arg name="model"/>
  <arg name="gui" default="False"/>
  <!--param name="robot_description" command="cat $(find robot_odometry)/URDF/rangerX.urdf"/>

  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" value="10.0"/>
  </node>
  <node pkg="joint_state_publisher" type="joint_state_publisher" name="joint_state_publisher">
    <param name="use_gui" value="True"/>
  </node-->

  <param name="serial_port" value="/dev/ttyUSB2" />
  <param name="angular_scale" value="0.4" /> <!-- max 0.5 -->
  <param name="linear_scale" value="0.43" /> <!-- max 0.5 -->
  <param name="left_power" value="60.0" /> <!-- left power factor -->
  <param name="right_power" value="60.0" /> <!-- right power factor -->
  <param name="min_power" value="15"/> <!--19, 17 avoiding, 15 approaching-->
  <param name="acceleration_speed" value="10" /> <!-- 80 -->
  <param name="acceleration_power" value="30" />
  <param name="deceleration_power" value="100" />

  <node pkg="morelab_robot_platform" type="eddie" name="eddie"/>
  <!--node pkg="morelab_robot_platform" type="eddie_ping" name="eddie_ping" />
  <node pkg="morelab_robot_platform" type="eddie_adc" name="eddie_adc"  /-->
  <node pkg="morelab_robot_platform" type="eddie_controller" name="eddie_controller"/>
  <node pkg="morelab_robot_platform" type="eddie_teleop" name="eddie_teleop" output="screen"/>
  <node pkg="morelab_robot_platform" type="eddie_odom" name="eddie_odom"/>

  <!--node pkg="rplidar_ros"  type="rplidarNode" name="rplidarNode">
    <param name="serial_port"         type="string" value="/dev/ttyUSB0"/>
    <param name="serial_baudrate"     type="int"    value="256000"/>
    <param name="frame_id"            type="string" value="laser"/>
    <param name="inverted"            type="bool"   value="false"/>
    <param name="angle_compensate"    type="bool"   value="true"/>
    <param name="scan_mode"           type="string" value="Standard"/>
  </node-->

  <node pkg="rviz" type="rviz" name="rviz"/>

  <param name="use_sim_time" value="false"/>
  <!--node pkg="gmapping" type="slam_gmapping" name="slam_gmapping">
    <param name="base_frame" value="base_footprint"/>
    <param name="map_frame" value="map"/>
    <param name="odom_frame" value="odom"/>
    <param name="xmin" value="-50.0"/>
    <param name="ymin" value="-50.0"/>
    <param name="xmax" value="50.0"/>
    <param name="ymax" value="50.0"/>
  </node-->

</launch>
